### PR TITLE
refactor(pandas): remove pre_execute

### PR DESCRIPTION
**Pull request by kszucs**
depends on #3489 and #3490 

BREAKING CHANGE: ibis.pandas.udf removed in favor of ibis.udf.vectorized

### CODE REVIEW

File: ibis/backends/dask/core.py [hunk 1]
```diff
@@ -262,21 +202,10 @@ def execute_until_in_scope(
...
-        **kwargs,
-    )
-
-    new_scope = scope.merge_scope(pre_executed_scope)
-
     # Short circuit: if pre_execute puts op in scope, then we don't need to
     # execute its computable_args
-    if new_scope.get_value(op, timecontext) is not None:
-        return new_scope
+    if scope.get_value(op, timecontext) is not None:
```
Comment (kszucs) [inline ibis/backends/dask/core.py:207]: Actually this should be removed.

File: ibis/backends/pandas/execution/udf.py [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+# 3) an ungrouped custom aggregation context
+@execute_node.register(
+    (ops.AnalyticVectorizedUDF, ops.ReductionVectorizedUDF),
+    pd.Series,
+    [pd.Series],
+)
+@execute_node.register(
+    (ops.AnalyticVectorizedUDF, ops.ReductionVectorizedUDF),
+    pd.Series,
+    pd.Series,
```
Comment (kszucs) [inline ibis/backends/pandas/execution/udf.py:59]: Required to resolve ambiguities with `(ops.Reduction, pd.Series, pd.Series)`

**Comment by cpcloud** [comment]
@jreback @gerrymanoim Does this affect you?

**Comment by cpcloud** [comment]
@kszucs I don't think we can remove this without an alternative. Probably better to wait on this for now.

**Comment by jreback** [comment]
deferring to @icexelloss on this, will comment shortly

**Comment by icexelloss** [comment]
> ibis.pandas.udf removed in favor of ibis.udf.vectorized
This is OK for us.

Removing pre_execute is problematic for us - we use that mechanism in quite a few places. We can look into removing all that usage but will take a bit, so perhaps not removing this now if it's not too problematic?

**Comment by cpcloud** [comment]
@kszucs Should we close this one out?

**Comment by kszucs** [comment]
Closing, I will be still able to re-open or cherry-pick from this PR.

**Pull request closed without merging**